sources/api/netdog/src/cli/install.rs

```diff
@@ -57,35 +55,26 @@ pub(crate) fn run(args: InstallArgs) -> Result<()> {
                 LeaseInfo::from_lease(&args.data_file).context(error::LeaseParseFailedSnafu {
                     path: &args.data_file,
                 })?;
-            // Randomize name server order, for libc implementations like musl that send
-            // queries to the first N servers.
-            let mut dns_servers: Vec<_> = info.dns_servers.iter().collect();
-            dns_servers.shuffle(&mut thread_rng());
-            write_resolv_conf(&dns_servers, &info.dns_search)?;
+
+            // Use DNS API settings if they exist, otherwise use DHCP lease
+            let dns_settings = if let Some(settings) =
+                DnsSettings::from_config().context(error::DnsFromConfSnafu)?
+            {
+                settings
+            } else {
+                DnsSettings::from_lease_info(&info)
+            };
+            dns_settings
+                .write_resolv_conf()
+                .context(error::ResolvConfWriteFailedSnafu)?;
```

Comment:
This relates to `from_config()` returning a valid  `DnsSettings` struct - potentially we want to divide this up rather than making it all API settings or all lease settings:
* if we have name-servers in the config, use those, otherwise use the lease
* if we have a search-list in the config, use it, otherwise use the lease